### ISSUE TITLE
Getting value from Textfield

### ISSUE THREAD

**Issue by berenddeboer** [aforemny/elm-mdc#68]
The slider component has an onChange callback, but I can't see such a thing with the Textfield. How do I update my model when the Textfield changes?

And what about blur and focus events?

**Comment by aforemny** [comment]
Hi, use `Options.onChange`, `Options.on`, etc. defined in `Material.Options`. Be sure to look at `Textfield.nativeControl`, too.

**Comment by berenddeboer** [comment]
One question though: why does the slider not use Material.Options.onChange?

**Comment by aforemny** [comment]
Hi @berenddeboer, the upstream implementation in MDC Web uses custom events for this component, too (last time I checked). This is related to the two facts (last time I checked):
- the Slider does not build upon a `nativeControl`, like TextField (`input`), etc.
- the Slider fires its events (change, input) from global `mousemove` events

Basically, when there are custom events being raised, we cannot do that with `Options.on*`. The Slider is the only component that does that. `Button.onClick` is another exception to the "always use `Options.on*`" rule.
